Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs

```diff
@@ -567,6 +573,7 @@ public void UpdateVisualState(DisposalUnitComponent component, bool flush)
             {
                 appearance.SetData(SharedDisposalUnitComponent.Visuals.VisualState, SharedDisposalUnitComponent.VisualState.Flushing);
                 appearance.SetData(SharedDisposalUnitComponent.Visuals.Light, SharedDisposalUnitComponent.LightState.Off);
+                component.Owner.SpawnTimer(component.GetFlushTime(), () => UpdateVisualState(component)); //update visuals once done flushing
```

Comment:
Don't use Timers please, they should be used only when really necessary and otherwise you should track time yourself (the main reason is this doesn't support gamestate serialization).